todo_app/github/github_authentication.py

```diff
@@ -0,0 +1,54 @@
+import os
+from todo_app.user.user import User
+import requests
+import random
+from hashlib import sha256
+
+class GithubAuthentication:
+    def __init__(self):
+        random.seed(a=None, version=2)
+        self._client_id = None
+        self._client_secret = None
+        self.state = sha256(str(random.random()).encode('ascii')).hexdigest()
+        self._authentication=None
+    
+
+    def get_client_id(self):
+        if not self._client_id:
+            self._client_id = os.environ["CLIENT_ID"]
+        return self._client_id
+    
+    def get_client_secret(self):
+        if not self._client_secret:
+            self._client_secret = os.environ["CLIENT_SECRET"]
+        return self._client_secret
+
+    def get_github_identity(self):
+        params={
+            "client_id": self.get_client_id(),
+            "redirect_uri": "http%3A%2F%2Flocalhost%2Flogin%2Fcallback",
```

Comment:
It's worth noting that this is optional - without it the app will use the default redirect_uri specified in Github's OAuth app config. That has the advantage of being easily configurable - this will redirect us incorrectly once this module is deployed to heroku, but if we relied on Github's config we could just use a different OAuth app for dev/prod
(Also note that this will only work for the app running locally on port 80 since no port is specified)